### PR TITLE
Feature/us17428 profile menu data

### PR DESCRIPTION
**Pull request by J-Byron**
- Updates shared-header/global-nav/profile-nav to accept external JSON

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -33,7 +33,7 @@
   <body>
 
     <div class="shared-header-skeleton" style="min-height: 49px; background-color: #0095d9">
-      <shared-header></shared-header>
+      <shared-header env="Int"></shared-header>
```
Comment (seancdavis) [inline src/index.html]: Note that this should be lower case. But let's leave it unchanged so it loads prod data.

File: src/components/shared-header/shared-header.tsx [hunk 2]
```diff
@@ -185,6 +185,7 @@ export class SharedHeader {
           giveNavIsShowing={this.giveNavIsShowing}
           navClickHandler={this.toggleMenu.bind(this)}
           giveData={this.data.give}
+          profileData={this.data.profile}
```
Comment (seancdavis) [inline src/components/shared-header/shared-header.tsx]: Where is this coming from? I don't see a profile key in crds-data.

File: src/components/shared-header/shared-header.tsx [hunk 3]
```diff
@@ -132,15 +132,15 @@ export class SharedHeader {
...
+        ? document.body.setAttribute('style', 'overflow: hidden; position: absolute; width: 100vw;')
         : document.body.setAttribute('style', 'overflow: scroll;');
     } else if (navType == 'give-nav') {
       this.giveNavIsShowing = !this.giveNavIsShowing;
       this.mainNavIsShowing = false;
       this.profileNavIsShowing = false;
       return this.giveNavIsShowing
         ? document.body.setAttribute('style', 'overflow: hidden; position: absolute; width: 100vw;')
-        : document.body.setAttribute('style', 'overflow: scroll;');
+        : document.body.setAttribute('style', 'overflow: scroll; ');
```
Comment (seancdavis) [inline src/components/shared-header/shared-header.tsx]: Why this change?

File: src/components/shared-header/profile-nav/profile-nav.tsx [hunk 4]
```diff
@@ -20,6 +21,48 @@ export class ProfileMenu {
...
+                <ul>
+                  {child.map(el => {
+                    if (typeof el != 'string')
+                      return (
+                        <li class={topLevel ? '' : 'top-level'}>
+                          <a
+                            href={el.path}
+                            data-automation-id={el['automation-id']}
+                            onClick={e => {
+                              if (el.title == 'Sign out') this.onSignOut(e);
```
Comment (seancdavis) [inline src/components/shared-header/profile-nav/profile-nav.tsx]: This is super brittle and feels like there's a little too much magic. If I change one character in "Sign out" then I lose the functionality? 

I'd rather see this as an option in the JSON object so I know what to expect.

File: src/components/shared-header/profile-nav/profile-nav.tsx [hunk 5]
```diff
@@ -20,6 +21,48 @@ export class ProfileMenu {
...
+          topLevel = topLevel || typeof child == 'string';
+
+          return (
+            <div style={{ padding: '0' }}>
+              {typeof child == 'string' && <h4>{child}</h4>}
+              {typeof child != 'string' && (
+                <ul>
+                  {child.map(el => {
+                    if (typeof el != 'string')
+                      return (
```
Comment (seancdavis) [inline src/components/shared-header/profile-nav/profile-nav.tsx]: This is really difficult to read. There are three return statements within this single function, and all contain a considerable amount of logic and data.

I'd break them out into their own functions so they are easier to read/understand.

File: src/components/shared-header/profile-nav/profile-nav.tsx [hunk 6]
```diff
@@ -20,6 +21,48 @@ export class ProfileMenu {
...
+    return (
+      <div>
+        <h2> {title} </h2>
+        {payload.children.map(child => {
+          topLevel = topLevel || typeof child == 'string';
+
+          return (
+            <div style={{ padding: '0' }}>
+              {typeof child == 'string' && <h4>{child}</h4>}
+              {typeof child != 'string' && (
```
Comment (seancdavis) [inline src/components/shared-header/profile-nav/profile-nav.tsx]: This is confusing. Breaking this out into a separate function can really help the readability here.

File: src/components/shared-header/profile-nav/profile-nav.tsx [hunk 7]
```diff
@@ -20,6 +21,48 @@ export class ProfileMenu {
...
 
+  renderSections = payload => {
+    let topLevel = false;
+
+    const title = payload.title.replace('%user_name%', this.currentUser.name.split(' ')[0]);
+
+    return (
+      <div>
+        <h2> {title} </h2>
+        {payload.children.map(child => {
```
Comment (seancdavis) [inline src/components/shared-header/profile-nav/profile-nav.tsx]: We've found it's a little easier to read when we save our maps for their own functions as opposed to injecting them directly into a return statement with other markup.

File: src/components/shared-header/profile-nav/profile-nav.tsx (hunk 4 above)
Comment (J-Byron) [reply src/components/shared-header/profile-nav/profile-nav.tsx]: Created a PR for adding signout key [here](https://github.com/crdschurch/crds-data/pull/5)
